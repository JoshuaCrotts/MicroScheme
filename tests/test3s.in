//
// Testing issue #3 bug.
//

void printlnd(int n) {
	printd(n);
	printc('\n');
}

void modify(int[] arr) {
	int i;
	int j = 2;
	int k;
	k = ++arr[j];
	arr[i] = ++arr[j] + arr[0];
	arr[0] = --arr[1];
	arr[1] = arr[2]++;
	arr[2] = arr[1]++ << (2 ** 3);
}

void main() {
	int arr[10];
	int i;
	for(i=0;i<#arr;i++)
		arr[i] = i;
	
	arr[0] = ++arr[1];
	arr[1] = arr[2]++ * 3;
	arr[1] = arr[0] + ++arr[1] + arr[0]++;
	
	for(i=0;i<#arr;i++)
		printlnd(--arr[i]);
	
	modify(arr);
	
	for(i=0;i<#arr;i++)
		printlnd(++arr[i] * arr[i]--);
}