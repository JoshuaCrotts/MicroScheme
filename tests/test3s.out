2
10
2
2
3
4
5
6
7
8
100
36
1050625
9
16
25
36
49
64
81
